src/core.js

```diff
@@ -229,7 +229,11 @@ function resumeProcessing( test ) {
 
 	// A slight delay to allow this iteration of the event loop to finish (more assertions, etc.)
 	if ( defined.setTimeout ) {
-		setTimeout( function() {
+		if ( test && test.resuming ) {
```

Comment:
This should guard the `begin()`, not the `setTimeout`. I believe the logic here will break down when a test's `assert.async()`s fail to resolve within 13ms of each other, e.g.
```js
QUnit.test( "fast and slow", function( assert ) {
	var done1 = assert.async(),
		done2 = assert.async();
	setTimeout( function() {
		assert.ok( true );
		done1();
	} );
	setTimeout( function() {
		assert.ok( true );
		done2();
	}, 100 );
} );
```